components/CMakeLists.txt

```diff
@@ -172,38 +176,62 @@ include_directories(${BULLET_INCLUDE_DIRS} ${CMAKE_CURRENT_BINARY_DIR})
 
 add_library(components STATIC ${COMPONENT_FILES} ${MOC_SRCS} ${ESM_UI_HDR})
 
-target_link_libraries(components 
-    ${Boost_SYSTEM_LIBRARY}
-    ${Boost_FILESYSTEM_LIBRARY}
-    ${Boost_THREAD_LIBRARY}
-    ${Boost_PROGRAM_OPTIONS_LIBRARY}
-    ${OSG_LIBRARIES}
-    ${OPENTHREADS_LIBRARIES}
-    ${OSGPARTICLE_LIBRARIES}
-    ${OSGUTIL_LIBRARIES}
-    ${OSGDB_LIBRARIES}
-    ${OSGVIEWER_LIBRARIES}
-    ${OSGGA_LIBRARIES}
-    ${OSGFX_LIBRARIES}
-    ${OSGANIMATION_LIBRARIES}
-    ${BULLET_LIBRARIES}
-    ${SDL2_LIBRARY}
-    # For MyGUI platform
-    ${OPENGL_gl_LIBRARY}
-    ${MYGUI_LIBRARIES}
-)
+if (NOT ANDROID)
+    target_link_libraries(components 
+        ${Boost_SYSTEM_LIBRARY}
+        ${Boost_FILESYSTEM_LIBRARY}
+        ${Boost_THREAD_LIBRARY}
+        ${Boost_PROGRAM_OPTIONS_LIBRARY}
+        ${OSG_LIBRARIES}
+        ${OPENTHREADS_LIBRARIES}
+        ${OSGPARTICLE_LIBRARIES}
+        ${OSGUTIL_LIBRARIES}
+        ${OSGDB_LIBRARIES}
+        ${OSGVIEWER_LIBRARIES}
+        ${OSGGA_LIBRARIES}
+        ${OSGFX_LIBRARIES}
+        ${OSGANIMATION_LIBRARIES}
+        ${BULLET_LIBRARIES}
+        ${SDL2_LIBRARY}
+        # For MyGUI platform
+        ${OPENGL_gl_LIBRARY}
+        ${MYGUI_LIBRARIES}
+        )
```

Comment:
Use an intermediate variable here to avoid duplicating the whole target_link_libraries. E.g.

<pre>
if (ANDROID)
   set(GL_LIB ${OPENGLES_gl_LIBRARY})
else()
   set(GL_LIB ${OpenGL_gl_LIBRARY})
endif()

target_link_libraries(
....
${GL_LIB}
...
)
</pre>